fintech-examples/fintech-ui/src/app/common/navbar/navbar.component.ts

```diff
@@ -1,39 +1,26 @@
 import { Component, OnInit } from '@angular/core';
 import { AuthService } from '../../services/auth.service';
-import { Consts } from '../../models/consts';
-import { Router } from '@angular/router';
-import { DocumentCookieService } from '../../services/document-cookie.service';
-import { LocalStorage } from '../../models/local-storage';
+import { StorageService } from '../../services/storage.service';
 
 @Component({
   selector: 'app-navbar',
   templateUrl: './navbar.component.html',
   styleUrls: ['./navbar.component.scss']
 })
 export class NavbarComponent implements OnInit {
-  constructor(private authService: AuthService, private cookieService: DocumentCookieService, private router: Router) {}
+  constructor(private authService: AuthService, private storageService: StorageService) {}
 
   ngOnInit() {}
 
   onLogout() {
-    this.authService.logout().subscribe(
-      ok => this.logOutHelper(),
-      notok => {
-        this.logOutHelper();
-      }
-    );
-  }
-
-  logOutHelper() {
-    LocalStorage.logout();
-    this.authService.openLoginPage();
+    this.authService.logout().subscribe();
```

Comment:
Hallo Collins, das hatte schon seinen Grund, dass ok und notOk auf die selbe Methode gingen. Im Falle von NotOk kann man sich jetzt nicht mehr ausloggen. Das ist Fatal. Insbesondere, wenn das am Donnerstag präsentiert werden soll. Bitte wieder umbauen.

Reply:
I used Promise instead of Observable to make sure the storage data are deleted after the logout request is done.

Reply:
I used Promise instead of Observable to make sure that the user always logs out after the logout request is done.